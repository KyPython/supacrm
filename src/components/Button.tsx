import React from "react";
import { IconType } from "react-icons";

type ButtonProps = React.ButtonHTMLAttributes<HTMLButtonElement> & {
  variant?: "primary" | "secondary" | "ghost";
  className?: string;
  leftIcon?: React.ReactNode | IconType;
  rightIcon?: React.ReactNode | IconType;
  ariaLabel?: string;
  href?: string;
};

function renderIcon(icon?: React.ReactNode | IconType) {
  if (!icon) return null;
  if (typeof icon === "function") {
    const C = icon as IconType;
    return (
      <span className="btn-icon" aria-hidden>
        <C />
      </span>
    );
  }
  return (
    <span className="btn-icon" aria-hidden>
      {icon}
    </span>
  );
}

export default function Button({
  children,
  variant = "primary",
  className = "",
  leftIcon,
  rightIcon,
  ariaLabel,
  href,
  ...props
}: ButtonProps) {
  const base =
    "inline-flex items-center justify-center gap-2 px-4 py-2 rounded focus:outline-none focus:ring-2 focus:ring-offset-2";

  const style: React.CSSProperties = {};
  if (variant === "primary") {
    style.background = "var(--brand)";
    style.color = "var(--fg)";
  }
  if (variant === "secondary") {
    style.background = "var(--card)";
    style.color = "var(--fg)";
  }

  const disabledClass = (props as any).disabled
    ? "opacity-60 cursor-not-allowed"
    : "hover:shadow-md";

  const iconLeft = renderIcon(leftIcon);
  const iconRight = renderIcon(rightIcon);

  if (href) {
    // anchor-style button
    return (
      <a
        aria-label={ariaLabel}
        className={`${base} ${disabledClass} ${className}`.trim()}
        style={style}
        href={href}
        {...(props as any)}
      >
        {iconLeft}
        <span>{children}</span>
        {iconRight}
      </a>
    );
  }

  return (
    <button
      aria-label={ariaLabel}
      className={`${base} ${disabledClass} ${className}`.trim()}
      style={style}
      {...props}
    >
      {iconLeft}
      <span>{children}</span>
      {iconRight}
    </button>
  );
}
