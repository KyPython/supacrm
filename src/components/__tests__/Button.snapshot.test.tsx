import { render } from "@testing-library/react";
import Button from "../Button";

test("Button snapshot", () => {
  const { container } = render(<Button variant="primary">Snapshot</Button>);
  expect(container).toMatchSnapshot();
});
