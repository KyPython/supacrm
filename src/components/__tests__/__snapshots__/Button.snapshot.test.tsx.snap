// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button snapshot 1`] = `
<div>
  <button
    class="inline-flex items-center justify-center gap-2 px-4 py-2 rounded focus:outline-none focus:ring-2 focus:ring-offset-2 hover:shadow-md"
  >
    <span>
      Snapshot
    </span>
  </button>
</div>
`;
